lsp/src/text_document.mli

```diff
@@ -2,7 +2,7 @@ open Types
 
 type t
 
-val make : DidOpenTextDocumentParams.t -> t
+val make : encoding:[ `UTF8 | `UTF16 ] -> DidOpenTextDocumentParams.t -> t
```

Comment:
1. I suppose so. It really only applies to updates anyway
2. This would be better if the encoding could be changed between updates but this isn't the case. So neither reflect the domain very well.

What would be a better long term fix is to general Document_store and promote it into the lsp library.